### PR TITLE
fix: bugfix on incr

### PR DESCRIPTION
**Pull request by neverchanje**
### What problem does this PR solve? 

https://github.com/apache/incubator-pegasus/pull/598 introduced a bug, which makes incr continue to proceed even the rocksdb::get failed.

### What is changed and how it works?

This PR fixes this bug and adds some unit tests using fail-point library to mock rocksdb get failure.

### Check List 

Tests 

- Unit test

### CODE REVIEW

File: src/server/pegasus_write_service_impl.h [hunk 1]
```diff
@@ -182,59 +182,61 @@ class pegasus_write_service::impl : public dsn::replication::replica_base
...
-        if (err == 0) {
-            if (!get_ctx.found) {
-                // old value is not found, set to 0 before increment
-                new_value = update.increment;
-                new_expire_ts = update.expire_ts_seconds > 0 ? update.expire_ts_seconds : 0;
-            } else if (get_ctx.expired) {
-                // ttl timeout, set to 0 before increment
-                _pfc_recent_expire_count->increment();
+        if (err != 0) {
+            resp.error = err;
```
Comment (Shuo-Jia) [inline src/server/pegasus_write_service_impl.h:186]: The line seem to be useless?

Comment (neverchanje) [reply src/server/pegasus_write_service_impl.h:186]: Certainly useful. It sets response error.

Comment (Shuo-Jia) [reply src/server/pegasus_write_service_impl.h:186]: But you `return error` , `resp.error` is not be used.

Comment (Shuo-Jia) [reply src/server/pegasus_write_service_impl.h:186]: Well, I konw

**Review by Shuo-Jia — APPROVED** [review]

**Review by acelyc111 — APPROVED** [review]

**Pull request merged**